KTO-1422 & KTO-1423 & KTO-358: Etusivun listojen parannuksia

pretseli opened the pull request:
Parannuksia koutan etusivun listojen suodattimiin:
- KTO-1422: Suodatus monella tilalla ("myös arkistoidut"-suodatin poistettu tarpeettomana)
- KTO-1423: Koulutustyyppi-suodatin
- KTO-358: Julkinen-sarake etusivun listoihin (koulutus ja valintaperuste)

Lisäksi refaktoroitu testit käyttämään elasticista dumpattua dataa ja poistettu testiriippuvuudet kouta-indeksoijaan, kouta-backendiin ja kouta-commoniin. Funktio kouta-index testidatan generointiin löytyy kouta-indeksoijasta: https://github.com/Opetushallitus/kouta-indeksoija/pull/190/files#diff-b0eaea2fa936bf7d48f396b4c30452f0714a02dd849550f3619a0c0cb65326f7

pitkamak reviewed (approved):
Olisiko testi-jsoneja mahdollista formatoida luettavampaan muotoon ? Vai rajoittaako elastic_dump jsonien muotoilua ?

pretseli committed:
Paremmat diffit testi-virheisiin

pretseli committed:
Korjattu api.clj sisennykset

pretseli committed:
KTO-439: Poistettu käyttämättömät filtered-list GET APIt

pretseli committed:
Lisätty koulutustyyppi-suodatin

pretseli committed:
KTO-358: Lisätty "julkinen"-kenttä koulutuksen ja valintaperusteen rajapintojen palauttamaan dataan

pretseli committed:
Refaktoroitu testit käyttämään elasticsearchista dumpattua dataa indeksoijan sijaan

- Poistettu riippuvuudet kouta-indeksoija, kouta-backend ja kouta-common

pretseli committed:
kouta-backend ja -indeksoija riippuvuuksien päivitys ja testien korjaus.

pretseli committed:
Muutettu "ei arkistoidut" testit "monella tilalla" testeiksi

pretseli committed:
Poistettu GET-rajapintojen testit (rajapinnat poistettu)

pretseli committed:
KTO-1423: Lisätty koulutustyyppi filters?-funktion ehtoon

pretseli committed:
Lisätty --add-host argumentti elasticdump-komentoon linuxia varten

pretseli committed:
Refaktoroitu ja paranneltu testejä

- Siirretty tyhjien indeksien testit omaan tiedostoon
- Siirretty post-200 apufunktiot test_tools-tiedostoon ja lisätty entiteetin nimi parametriksi, jotta voidaan käyttää kaikissa testeissä
- Ladataan elasticsearch dumppi testeissä vain jos elasticsearch on tyhjä.

pretseli committed:
KTO-358: Lisätty "julkinen" sorttauksessa käytettäviin kenttiin

pretseli committed:
KTO-439: Poistettu erillinen arkistoidut-filtteröinti ja muutettu tila-suodatin ottamaan lista tiloja

pretseli committed:
Lisätty koulutus filtered-list testi koulutustyyppi-parametrille

pretseli committed:
Ajetaan elasticdump dockerissa

pretseli committed:
.clj-kondo gitignoreen

heidilm commented on test/kouta_index/filtered_haku_list_test_post.clj:
```diff
@@ -1,147 +1,96 @@
-            [kouta-index.test-tools :as tools]
-            [kouta-indeksoija-service.fixture.external-services :as mocks]))
+            [kouta-index.test-tools :refer :all]
+            ;[kouta-index.prepare-indexed-data :refer :all]
```
Onko tämä tarkoituksella kommentoitu pois?

pretseli replied:
Oikeastaan tän vois poistaa kokonaan. Tuo on siirretty indeksoidaan.

pretseli commented:
> Olisiko testi-jsoneja mahdollista formatoida luettavampaan muotoon ? Vai rajoittaako elastic_dump jsonien muotoilua?

"The file format generated by this tool is line-delimited JSON files. The dump file itself is not valid JSON, but each line is. We do this so that dumpfiles can be streamed and appended without worrying about whole-file parser integrety."

pretseli committed:
Poistettu kommentoitu require
